pkg/webhook/pod/mutating/sidecarset.go

```diff
@@ -64,13 +67,28 @@ func (h *PodCreateHandler) sidecarsetMutatingPod(ctx context.Context, req admiss
 	}
 
 	// DisableDeepCopy:true, indicates must be deep copy before update sidecarSet objection
+
 	sidecarsetList := &appsv1alpha1.SidecarSetList{}
-	if err = h.Client.List(ctx, sidecarsetList, utilclient.DisableDeepCopy); err != nil {
+	sidecarsetList2 := &appsv1alpha1.SidecarSetList{}
+	podNamespace := pod.Namespace
+	if podNamespace == "" {
+		podNamespace = "default"
+	}
+	if err := h.Client.List(ctx, sidecarsetList, client.MatchingFields{fieldindex.IndexNameForSideCarSetNamespace: podNamespace}, utilclient.DisableDeepCopy); err != nil {
 		return false, err
 	}
-
+	if err := h.Client.List(ctx, sidecarsetList2, client.MatchingFields{fieldindex.IndexNameForSideCarSetNamespace: fieldindex.IndexValueSideCarSetClusterScope}, utilclient.DisableDeepCopy); err != nil {
+		return false, err
+	}
+	handleSideCarSetMap := make(map[string]bool, len(sidecarsetList.Items)+len(sidecarsetList2.Items))
 	matchedSidecarSets := make([]sidecarcontrol.SidecarControl, 0)
-	for _, sidecarSet := range sidecarsetList.Items {
+	for _, sidecarSet := range append(sidecarsetList.Items, sidecarsetList2.Items...) {
+		// Prevents sidecarSet from processing twice
+		key := fmt.Sprintf("%s/%s", sidecarSet.Namespace, sidecarSet.Name)
```

Comment:
sidecarset is a clusterscope resource, so the name of sidecarset should be unique clusterwise, it is not necessary to add namespace in the key. Actually it seems impossible for a sidecarset to be both cluster scoped and namespace scoped, so is it really possible to precess sidecarset twice here?